### ISSUE TITLE
Add filename_suffix option to ModelCheckpoint

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/ignite#129]
I would like to add `filename_suffix` option to ModelCheckpoint as discussed in [here](https://github.com/pytorch/ignite/pull/75). The code is [here](https://github.com/vfdev-5/ignite/commit/f374ea83280cbbed5ed752e1e08cf9aa859ef33c).

Idea is to be able to store a score in the filename. For example, if we specify `filename_suffix="_val_loss={score:.4f}"` and `score_function` that returns `-loss`, saved models filenames will be, for example, `model_SqueezeNet_98_val_loss=1.2413.pth`. See [this example](https://github.com/vfdev-5/ignite/blob/new_examples/examples/cifar10_playground.py#L226).

@alykhantejani @jasonkriss what do you think ?

**Comment by elanmart** [comment]
@vfdev-5 I'm sorry for not responding in the PR thread, I was away from the internet for few weeks. 

Personally I don't have a strong need for this feature, but I think that storing the priority can be useful.

I'm not sure about the API though. `filename_suffix="_val_loss={score:.4f}"` seems too implicit and I would surely mess that up.

How about providing just the suffix name? 

```python
if suffix is not None:
    full_suffix = f'{suffix}={abs(priority)}'
    fname = '{}_{}_{}_{}.pth'.format(self._fname_prefix, name, self._iteration, full_suffix)
else:
    fname = '{}_{}_{}.pth'.format(self._fname_prefix, name, self._iteration)

**Comment by alykhantejani** [comment]
I agree with @elanmart about the `_val_loss={score.4f}` being too complicated. I noticed that the current version will already add the loss / epoch count to the filename (see [here](https://github.com/pytorch/ignite/blob/master/ignite/handlers/checkpoint.py#L142)).

Perhaps adding a `count_or_score_prefix` (with hopefully a better name) and [this line](https://github.com/pytorch/ignite/blob/master/ignite/handlers/checkpoint.py#L142) can just change to 
```python
fname = '{}_{}_{}_{}.pth'.format(self._fname_prefix, name, self._count_or_score_prefix, self._iteration)
```

**Comment by vfdev-5** [comment]
@elanmart 👍 nice with f-string :)

> I noticed that the current version will already add the loss / epoch count to the filename (see here).

@alykhantejani could you please explain ?

**Comment by alykhantejani** [comment]
@vfdev Ah sorry I misread a line in the source. What I meant was 

The current output filename is `fname = '{}_{}_{}.pth'.format(self._fname_prefix, name, self._iteration)` instead we could swap this to `fname = '{}_{}_{}.pth'.format(self._fname_prefix, name, priority)` as `priority` is set to `self._iteration` if no `self._score_ function` is not defined.

So then if your score/loss is -1.287 the output would be `prefix_name_-1.287.pth`. Additionally, we can add a prefix to this score by doing what @elanmart said.

**Comment by vfdev-5** [comment]
@alykhantejani your solution is more simple 👍 , but in terms of `prefix` and `name` can look like a hack: 
`prefix="resnet"` and `name="val_loss"` that gives "resnet_val_loss_-1.287.pth". Maybe we need redefine `prefix -> name` and `name - > suffix` ?

**Comment by elanmart** [comment]
@alykhantejani actually I think it's better to have both `priorty` and `iteration`, than only `priority`. Then the filename tells you
how long you should run your model to get the same weights.

**Comment by vfdev-5** [comment]
@alykhantejani yep, will do very very soon

**Comment by alykhantejani** [comment]
merged in #132
